app/src/androidTest/java/com/github/wnder/TourDatabaseTest.java

```diff
@@ -0,0 +1,199 @@
+import com.github.wnder.tour.FirebaseTourDatabase;
+import com.github.wnder.tour.TourDatabase;
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.firestore.FirebaseFirestore;
+import com.google.firebase.storage.FirebaseStorage;
+import com.mapbox.mapboxsdk.geometry.LatLng;
+
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+
+import static java.lang.Thread.sleep;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+
+@RunWith(JUnit4.class)
+public class TourDatabaseTest {
+
+    private static FirebaseTourDatabase tdb;
+    private static FirebasePicturesDatabase db;
+    private static String tourUniqueId;
+    private static String tourName;
+    private static String firstUniqueId;
+    private static String secondUniqueId;
+    private static String thirdUniqueId;
+    private static List<String> pictures;
+    private static Location firstLoc;
+    private static Location secondLoc;
+    private static Location thirdLoc;
+    private static double totalLength;
+    private static String user;
+
+    @BeforeClass
+    public static void createTestTour() {
+
+        tdb = new FirebaseTourDatabase(ApplicationProvider.getApplicationContext());
+        db = new FirebasePicturesDatabase(ApplicationProvider.getApplicationContext());
+
+        firstLoc = new Location("");
+        firstLoc.setLatitude(0);
+        firstLoc.setLongitude(0);
+
+        secondLoc = new Location("");
+        secondLoc.setLatitude(10);
+        secondLoc.setLongitude(1);
+
+        thirdLoc = new Location("");
+        thirdLoc.setLatitude(5);
+        thirdLoc.setLongitude(2);
+
+        user = "testUser";
+        firstUniqueId = "first" + user + Calendar.getInstance().getTimeInMillis();
+        secondUniqueId = "second" + user + Calendar.getInstance().getTimeInMillis();
+        thirdUniqueId = "third" + user + Calendar.getInstance().getTimeInMillis();
+
+        CompletableFuture<Void> uploadFirstPic = db.uploadPicture(firstUniqueId, new UploadInfo(user, firstLoc, Uri.parse("android.resource://com.github.wnder/" + R.raw.ladiag)));
+        CompletableFuture<Void> uploadSecondPic = db.uploadPicture(secondUniqueId, new UploadInfo(user, secondLoc, Uri.parse("android.resource://com.github.wnder/" + R.raw.ladiag)));
+        CompletableFuture<Void> uploadThirdPic = db.uploadPicture(thirdUniqueId, new UploadInfo(user, thirdLoc, Uri.parse("android.resource://com.github.wnder/" + R.raw.ladiag)));
+
+        try {
+            uploadFirstPic.get();
+            uploadSecondPic.get();
+            uploadThirdPic.get();
+        } catch (Exception e) {
+            e.printStackTrace();
```

Comment:
The exception should be added to the test method signature so that it always fail in case an exception is thrown

Reply:
done for all tests